shared/AppInsightsCore/src/JavaScriptSDK/CoreUtils.ts

```diff
@@ -146,6 +183,63 @@ export function objForEachKey(target: any, callbackfn: (name: string, value: any
     }
 }
 
+/**
+ * Effectively assigns all enumerable properties (not just own properties) and functions (including inherited prototype) from 
+ * the source object to the target, it attempts to use proxy getters / setters (if possible) and proxy functions to avoid potential
+ * implementation issues by assigning prototype functions as instance ones
+ * 
+ * This method is the primary method used to "update" the snippet proxy with the ultimate implementations.
+ * 
+ * Special ES3 Notes:
+ * Updates (setting) of direct property values on the target or indirectly on the source object WILL NOT WORK PROPERLY, updates to the 
+ * properties of "referenced" object will work (target.context.newValue = 10 => will be reflected in the source.context as it's the
+ * same object). ES3 Failures: assigning target.myProp = 3 -> Won't change source.myProp = 3, likewise the reverse would also fail.
+ * @param target - The target object to be assigned with the source properties and functions
+ * @param source - The source object which will be assigned / called by setting / calling the targets proxies
+ * @param chkSet - An optional callback to determine whether a specific property/function should be proxied
+ * @memberof Initialization
+ */
+export function proxyAssign(target: any, source: any, chkSet?: (name: string, isFunc?: boolean, source?: any, target?: any) => boolean) {
```

Comment:
Helper used for snippet initialization, uses proxy functions, object defineProperty() so the snippet will now "act" as a proxy for the underlying instance, previously we effectively "morphed" the snippet into final object by overwriting ALL properties and prototype methods -- not ideal and potentially fragile for future JS runtimes.